processor/src/trace/virtual_bus/circuit/mod.rs

```diff
@@ -11,8 +14,83 @@ pub use prover::prove;
 mod verifier;
 pub use verifier::verify;
 
+use super::multilinear::inner_product;
 use super::sum_check::{FinalOpeningClaim, Proof as SumCheckProof};
 
+/// Defines the number of elements for the partial left/right numerator/denominators of
+/// [`LayerGatesInputs`].
+const NUM_ELEMENTS_PER_GATE_INPUT: usize = 4;
```

Comment:
nit: probably `1 <<  2` to emphasis the need for things to be a power of two.

Reply:
Good idea; but added a static assertion using [`static_assertions`](https://docs.rs/static_assertions/latest/static_assertions/) instead